phoenix-core/src/main/java/org/apache/phoenix/filter/AllVersionsIndexRebuildFilter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.filter;
+
+import java.io.IOException;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.filter.Filter;
+
+/**
+ * This filter overrides the behavior of delegate so that we do not jump to the next
+ * column as soon as we find a value for a column but rather include all versions which is
+ * needed for rebuilds.
+ */
+public class AllVersionsIndexRebuildFilter extends DelegateFilter {
+
+    public AllVersionsIndexRebuildFilter(Filter originalFilter) {
+        super(originalFilter);
+    }
+
+    @Override
+    public ReturnCode filterKeyValue(Cell v) throws IOException {
+        ReturnCode delegateCode = super.filterKeyValue(v);
+        if (delegateCode == ReturnCode.INCLUDE_AND_NEXT_COL) {
```

Comment:
Could you add a comment here why we convert it to INCLUDE? Why are we not happy with NEXT_COL?

Reply:
@gokceni NEXT_COL skips this column and goes to the next one. What we want to do is when the underlying filter says yes to a column, we want to say yes too, but instead of jumping to the next col since we got a value, we want to get all versions.

Reply:
Yes, I agree, comment in the code would be helpful here. @abhishek-chouhan